internal/records/events.go

```diff
@@ -1,11 +1,30 @@
 package records
 
 import (
+	"context"
 	"time"
+
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
 )
 
+var log = logf.Log.WithName("events")
+
 const JobSetRecordsAnnotationKey = "megamon.tbd/records"
 
+type contextType int
```

Comment:
Passing specific values through a context should not be needed. You can pass the logger through the context. The logger have values attached.